catalog/transformer/catalog-transformer-tika-input/src/main/java/ddf/catalog/transformer/input/tika/TikaInputTransformer.java

```diff
@@ -506,14 +506,10 @@ protected MetacardType mergeAttributes(MetacardType metacardType) {
             .flatMap(Collection::stream)
             .collect(Collectors.toSet());
 
-    // Guard against empty collection. If the collection is empty,
-    // the MetacardTypeImpl constructor throws an exception.
-    if (!additionalAttributes.isEmpty()) {
-      returnObject =
-          new MetacardTypeImpl(metacardType.getName(), metacardType, additionalAttributes);
-    }
+    // always add extracted attributes
+    additionalAttributes.addAll(new ExtractedAttributes().getAttributeDescriptors());
```

Comment:
❓ I'm not sure I understand why we need to use the `ExtractedAttribute` class now to get the text. Didn't we get the text before? If so, what changed to cause this problem?

Reply:
The text won't be persisted if it isn't in the metacard type descriptors. This can happen when an Extractor is providing the metacard type but doesn't include this attribute. This ensures that whatever the passed in type we will define the attribute so it will be persisted.

Reply:
I guess I'm trying to understand what changed. I assumed that the text used to be extracted and persisted before, or wasn't it the case?

Reply:
This changed.

Before the text was being stored in the metacard's metadata attribute.
This was recently changed so that the document's metadata is stored in the metacard's metadata attribute and the body of the document is store (up to a configurable limit) in the new ext.extracted.text attribute.

Reply:
What @gordocanchola said

Reply:
Ok, thanks for the clarification!